helpers/compile/plugins/replaceWithPlugin.ts

```diff
@@ -14,15 +21,44 @@ function applyReplacements(contents: string, replacements: [RegExp, string][]) {
  * @param replacements
  * @returns
  */
-export const replaceWithPlugin = (replacements: [RegExp, string][]): esbuild.Plugin => {
+export const replaceWithPlugin = (replacements: Replacement[]): esbuild.Plugin => {
   return {
     name: 'replaceWithPlugin',
     setup(build) {
       build.onLoad({ filter: /.*/ }, async (args) => {
+        // we skip, don't attempt to edit files that aren't js
+        if (builtinModules.includes(args.path)) return {}
+        if (!/.*?(.js|.mjs)$/.exec(args.path)) return {}
+
         const contents = await fs.promises.readFile(args.path, 'utf8')
 
-        return { contents: applyReplacements(contents, replacements) }
+        return { contents: await applyReplacements(contents, replacements) }
       })
     },
   }
 }
+
+/**
+ * Example
+ */
+// const inlineUndiciWasm = replaceWithPlugin([
+//   [
+//     /(await WebAssembly\.compile\().*?'(.*?)'\)\)\)/g,
+//     async (regex, contents) => {
+//       for (const match of contents.matchAll(regex)) {
+//         if (match[2].includes('simd') === false) {
+//           // we only bundle lhttp wasm files that are not simd compiled
+//           const engineCoreDir = resolve.sync('@prisma/engine-core')
+//           const undiciPackage = resolve.sync('undici/package.json', { basedir: engineCoreDir })
+//           const lhttpWasmPath = path.join(path.dirname(undiciPackage), 'lib', match[2])
+//           const wasmContents = (await fs.promises.readFile(lhttpWasmPath)).toString('base64')
+//           const inlineWasm = `${match[1]}(Buffer.from("${wasmContents}", "base64")))`
+
+//           contents = contents.replace(match[0], inlineWasm)
+//         }
+//       }
+
+//       return contents
+//     },
+//   ],
+// ])
```

Comment:
What is this example for @millsp ?

Reply:
It used to show how to inline wasm when the wasm file is not directly imported. However, in the mean time undici changed how they load the wasm and now puts the wasm contents in a .js file instead of doing a `readFileSync` so this is not needed anymore and works well when bundled.

Reply:
Should we remove this then in a follow up PR?

Reply:
We can probably trim it a bit, but I'd like to keep an example for how to use `replaceWithPlugin`.